litmus-portal/cluster-agents/subscriber/pkg/cluster/events/util.go

```diff
@@ -17,44 +19,61 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/serializer/yaml"
 )
 
-// util function, extracts the chaos data using the litmus go-client
-func getChaosData(nodeStatus v1alpha13.NodeStatus, engineName, engineNS string, chaosClient *v1alpha12.LitmuschaosV1alpha1Client) (*types.ChaosData, error) {
+// 0 means no resync
+const (
+	resyncPeriod time.Duration = 0
+)
+
+var (
+	AgentScope     = os.Getenv("AGENT_SCOPE")
+	AgentNamespace = os.Getenv("AGENT_NAMESPACE")
+	ClusterID      = os.Getenv("CLUSTER_ID")
+)
+
+// GetChaosData is a util function, extracts the chaos data using the litmus go-client
+func GetChaosData(nodeStatus v1alpha13.NodeStatus, engineName, engineNS string, chaosClient *v1alpha12.LitmuschaosV1alpha1Client) (*types.ChaosData, error) {
 	cd := &types.ChaosData{}
 	cd.EngineName = engineName
 	cd.Namespace = engineNS
 	crd, err := chaosClient.ChaosEngines(cd.Namespace).Get(cd.EngineName, v1.GetOptions{})
 	if err != nil {
 		return nil, err
 	}
-	// considering chaos engine has only 1 experiment
+	if nodeStatus.StartedAt.Unix() > crd.ObjectMeta.CreationTimestamp.Unix() {
+		return nil, errors.New("chaosenginge resource older than current workflow node")
+	}
+	cd.ProbeSuccessPercentage = "0"
+	cd.FailStep = ""
+	cd.EngineUID = string(crd.ObjectMeta.UID)
+	if len(crd.Status.Experiments) == 0 {
+		return cd, nil
+	}
+	// considering chaosengine will only have 1 experiment
+	cd.ExperimentPod = crd.Status.Experiments[0].ExpPod
+	cd.RunnerPod = crd.Status.Experiments[0].Runner
+	cd.ExperimentStatus = string(crd.Status.Experiments[0].Status)
+	cd.ExperimentName = crd.Status.Experiments[0].Name
+	cd.LastUpdatedAt = strconv.FormatInt(crd.Status.Experiments[0].LastUpdateTime.Unix(), 10)
+	cd.ExperimentVerdict = crd.Status.Experiments[0].Verdict
+	if strings.ToLower(string(crd.Status.EngineStatus)) == "stopped" || (strings.ToLower(string(crd.Status.EngineStatus)) == "completed" && strings.ToLower(cd.ExperimentVerdict) != "pass") {
+		cd.ExperimentVerdict = "Fail"
+		cd.ExperimentStatus = "Stopped"
```

Comment:
Is ExperimentStatus should be stopped always for both stopped and completed cases?
or it should be cd.ExperimentStatus = string(crd.Status.EngineStatus)